mount/temp.go

```diff
@@ -58,6 +58,24 @@ func WithTempMount(ctx context.Context, mounts []Mount, f func(root string) erro
 			}
 		}
 	}()
+
+	// The volatile option of overlayfs doesn't allow to mount again using the
+	// same upper / work dirs. Since it's a temp mount, avoid using that
+	// option here.
+	// TODO: Make this logic conditional once the kernel supports reusing
+	// overlayfs volatile mounts.
+	for _, m := range mounts {
```

Comment:
Would you please add that reference https://docs.kernel.org/filesystems/overlayfs.html#volatile-mount in that comment and add that test in `mount` package? thanks

NOTE: Currently, containerd uses read-only mount to get uid/gid. But containerd still needs to strip out volatile option on temp mounts.

* https://github.com/containerd/containerd/pull/6478/files
* https://github.com/containerd/containerd/pull/8259

Reply:
@fuweid addressed your comments! PTAL